Fix Custom Tabs implementation guide (#4402)
GoogleChrome/developer.chrome.com: site/en/docs/android/custom-tabs/integration-guide/index.md

By default, Custom Tabs launch as a full-window activity. Starting in Chrome 107, you can use partial Custom Tabs to specify a different launch height such that users can interact with your app while viewing web content. Users will be able to expand the Custom Tab to full-screen by dragging the toolbar handle up and restoring the initial launch height by dragging the handle down.

{% Img src="image/6AZNJBRnkpQUWTKPzig99lQY8jT2/iMTvWNK7aEqqxHEHEXuv.png", alt="Example Partial Tab", width="584", height="1168", class="screenshot" %}

You can also choose to make the Custom Tab non-resizable. In this case, users will not have the ability to resize the Custom Tab. If you set a value below 50% of the screen height in the intent, Chrome automatically adjusts the Custom Tab to 50% of the screen height. Custom Tabs are not supported when users are in multi-window or landscape mode.

Partial Custom Tabs are supported by the AndroidX browser library from version 1.5 onwards:

```groovy
implementation 'androidx.browser:browser:1.5.0-alpha01'
```

> To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

Author: The hash is intended. It's the javadoc notation for referencing a class member.

> To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

Reviewer: What does a developer actually type in a program? A hash or a period?

> To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

Author: A period, my theory is that they use the hash as a separator between class members and nested classes. Here is the corresponding Javadoc spec: https://docs.oracle.com/en/java/javase/17/docs/specs/javadoc/doc-comment-spec.html#link

> To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

Reviewer: Then this needs to be a period. Developer documentation has to show what the developer actually does and not require them to interpret a convention they may or may not be familiar with. Being instantly understandable is better than providing a link to an explanation.

> To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

Author: The developer will never call `CustomTabsIntent.Builder.setInitialActivityHeightPx()` as it is not a static method, but instead will have to call it on an instance like this `myBuilder.setInitialActivityHeightPx(). This is really just a pointer to the actual method (I would've added a link, but the reference docs are not yet published). 

We can change it to `new CustomTabsIntent.Builder().setInitialActivityHeightPx(heightInPx)`  if that helps.

> To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

Author: rephrased to `new CustomTabsIntent.Builder().setInitialActivityHeightPx(heightInPx)`

The snippet below changes the launch height of the Custom Tab. `CustomTabsIntent.Builder` has a new method `setInitialActivityHeightPx()` that allows you to define the initial launch height in pixels. As a prerequisite to using Custom Tabs, you will need to [connect to the Custom Tabs Service](/docs/android/custom-tabs/integration-guide/#connect-to-the-custom-tabs-service) or launch the custom tabs intent using [Activity#startActivityForResult()](/reference/android/app/Activity#startActivityForResult(android.content.Intent,%20int)).
To turn a Custom Tab into a partial Custom Tab, define the initial launch height in pixels by calling `CustomTabsIntent.Builder#setInitialActivityHeightPx()`. Furthermore, you need to either:

> There are properties that do not work when building an Intent for partial Custom Tabs:

Reviewer 2: Delete this line too? (see comment farther down)

Author: this is removed

> Starting a new browser sessions is the recommended approach to launching a partial Custom Tag, as it enables you to [warm up the browser process](#warm-up-the-browser-process) and [pre-render webpages](/docs/android/custom-tabs/best-practices/#pre-render-content). If you don't need these features, using `startActivityForResult()` is the easier way to launch a partial custom tab.

Reviewer 2: ```suggestion
Starting a new browser session is the recommended approach to launching a partial Custom Tab, as it enables you to [warm up the browser process](#warm-up-the-browser-process) and [pre-render webpages](/docs/android/custom-tabs/best-practices/#pre-render-content). If you don't need these features, using `startActivityForResult()` is the easier way to launch a partial Custom Tab.
```

> Starting a new browser sessions is the recommended approach to launch a partial Custom Tag, as it enables you to [warm up the browser process](#warm-up-the-browser-process) and [pre-render webpages](/docs/android/custom-tabs/best-practices/#pre-render-content). If you don't need these features, using `startActivityForResult` is the easier way to launch a partial custom tab.

Reviewer: ```suggestion
Starting a new browser sessions is the recommended approach to launching a partial Custom Tag, as it enables you to [warm up the browser process](#warm-up-the-browser-process) and [pre-render webpages](/docs/android/custom-tabs/best-practices/#pre-render-content). If you don't need these features, using `startActivityForResult()` is the easier way to launch a partial custom tab.
```

{% Aside 'gotchas' %}
* Specifying the initial activity height will not have an effect if the default browser does not support resizing the Custom Tab. In this case, the intent extra will be ignored and the Custom Tab will span the complete display height.

> * Custom Tabs will inherit the host app's color scheme for the user interface properties above. You will be responsible for ensuring visual consistency for these properties before launching a Custom Tab. This means the `CustomTabColorScheme.navigationBarColor` and `CustomTabColorScheme.navigationBarDividerColor` properties do not work when building an intent for a Custom Tab.

Reviewer 2: Could be because I'm new the review UI, but it looks like the lines listing the properties above got deleted, which is fine. We can just list them here but need to rephrase.

Author: Good catch! Added "for a partial Custom Tab"

```java
CustomTabsSession customTabsSession;

// ...

CustomTabsIntent customTabsIntent = new CustomTabsIntent.Builder(session)
  .setInitialActivityHeightPx(500)
  .setCloseButtonPosition(CustomTabsIntent.CLOSE_BUTTON_POSITION_END)
  // ...
  .build();

customTabsIntent.launchUrl(context, Uri.parse(url))
```

### Launch a partial custom tab via startActivityForResult

```java
private ActivityResultLauncher<String> mCustomTabLauncher = registerForActivityResult(new ActivityResultContract<String, Integer>() {
    @Override
    public Integer parseResult(int statusCode, @Nullable Intent intent) {
        return statusCode;
    }

    @NonNull
    @Override
    public Intent createIntent(@NonNull Context context, String url) {
        CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder()
                .setInitialActivityHeightPx(500)
                .setCloseButtonPosition(CustomTabsIntent.CLOSE_BUTTON_POSITION_END)
                .setToolbarCornerRadiusDp(10);
        Intent customTabsIntent = builder.build().intent;
        customTabsIntent.setData(Uri.parse(url));
        return customTabsIntent;
    }
}, new ActivityResultCallback<Integer>() {
    @Override

>     public void onActivityResult(Integer statusCode) {

Reviewer 2: Is it worth noting that Chrome is actually going to return a result?

Author: Absolutely, I've been wondering that as well, what is it going to return?

boolean ok = CustomTabsClient.bindCustomTabsService(this, mPackageNameToBind, connection);
```

> When targeting API level 30, in order to connect to a custom tabs service, you need to add a queries section to your Android Manifest, declaring an intent-filter that matches browsers with Custom Tabs support.

Reviewer 2: ```suggestion
When targeting API level 30, in order to connect to a Custom Tabs service, you need to add a queries section to your Android Manifest, declaring an intent-filter that matches browsers with Custom Tabs support.
```

Reviewer 2: Wondering in general if we want "Custom Tab" to be capitalized throughout?
